clippy_lints/src/functions.rs

```diff
@@ -411,6 +450,36 @@ impl<'tcx> Functions {
     }
 }
 
+fn check_result_unit_err(cx: &LateContext<'_>, decl: &hir::FnDecl<'_>, item_span: Span, fn_header_span: Span) {
+    if in_external_macro(cx.sess(), item_span) {
+        return;
+    }
+    if let hir::FnRetTy::Return(ref ty) = decl.output {
```

Comment:
Most of the codebase uses `if_chain` with more than two `if`/`if let`. Although I agree it's a pain that rustfmt and rust-analyzer don't work there, I would prefer to keep the consistency.

Reply:
Good idea. I didn't introduce `if_chain!` here at first because the rightwards drift didn't lead to hard-to-read formatting, but now I have added it, I feel it reads even better.